sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -486,11 +486,35 @@ def _get_index(self):
 
   first = last = frame_base.wont_implement_method('order-sensitive')
   head = tail = frame_base.wont_implement_method('order-sensitive')
+  interpolate = frame_base.wont_implement_method('order-sensitive')
 
 
 @populate_not_implemented(pd.Series)
 @frame_base.DeferredFrame._register_for(pd.Series)
 class DeferredSeries(DeferredDataFrameOrSeries):
+  @property
+  def name(self):
+    return self._expr.proxy().name
```

Comment:
Another implementation question: does the return proxy() change over this DeferredSeries lifetime?

Reply:
The `proxy()` for an expression (`self._expr`) should never change.

In general the `_expr`  for a `DeferredFrame` does not change, but it can happen, for example when performing an inplace operation:
https://github.com/apache/beam/blob/a8cd05932bed9b2480316fb8518409636cb2733b/sdks/python/apache_beam/dataframe/frame_base.py#L328-L337

So if you access `name` before and after an inplace operation it will point to a different proxy.

Reply:
Good to know! Thanks